src/nvim/getchar.c

```diff
@@ -3182,3 +3189,121 @@ bool map_execute_lua(bool may_repeat)
   ga_clear(&line_ga);
   return true;
 }
+
+/// Appends pasted text to redo buffer and/or record buffer if needed.
+/// Escapes all K_SPECIAL and NUL bytes in the content.
+///
+/// @param state  kFalse for the start of a paste
+///               kTrue for the end of a paste
+///               kNone for the content of a paste
+/// @param str    the content of the paste (only used when state is kNone)
+void store_pasted_text(const TriState state, const String str, const bool crlf)
```

Comment:
is it worth using a common prefix?
```suggestion
void paste_store(const TriState state, const String str, const bool crlf)
```

Reply:
Yes